sponsorship_switzerland/models/completion_rules.py

```diff
@@ -79,9 +80,23 @@ def get_from_partner_ref(self, stmts_vals, st_line):
         if 'ref' in st_line:
             ref = st_line['ref']
         res = {}
+
+        partner_ref = ref[9:16]
         partner_obj = self.env['res.partner']
-        partner = partner_obj.search(
-            [('ref', '=', str(int(ref[9:16])))])
+        partner = partner_obj.search([('ref', '=', str(int(partner_ref)))])
+        if not partner:
+            # Some bvr reference have a wrong number of leading zeros,
+            # resulting in the partner reference to be offset.
+            flexible_ref_match = re.search(r'^0{,10}([1-9]\d{6})0',
```

Comment:
Not sure what you are doing here. Is it correct the following understanding:

- Up to ten 0
- One digit
- Six digits
- One 0

If that's correct I don't understand the middle part with the digits? [1-9] and \d are the same, no?

Reply:
[1-9] doesn't include 0. This is useful to separate the leading zeros to the partner id (that cannot start with a 0).

Reply:
There are some smaller refs as well, would it be more risky to change to this?
```suggestion
            flexible_ref_match = re.search(r'^0{,10}([1-9]\d{1,6})0',
```